honeybee/door.py

```diff
@@ -73,19 +97,46 @@ def from_vertices(cls, name, vertices, boundary_condition=None):
 
     @property
     def name(self):
-        """The door name (including only legal characters)."""
+        """Get or set the door name (including only legal characters)."""
         return self._name
 
+    @name.setter
```

Comment:
It looks like we can take advantage of sub-classing in this library. We can break down the objects to `Base` and `BaseWithShading` and add name, type, etc to base-class.

Reply:
`BaseWithShading` sounds much better.  Ok. I'll re-vamp all of the objects now.  It looks like `name` and `geometry` are really the only shared properties but I can still see it saves us a lot of redundant code.